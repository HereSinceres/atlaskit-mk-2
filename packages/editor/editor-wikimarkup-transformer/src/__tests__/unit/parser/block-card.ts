import WikiMarkupTransformer from '../../../index';

describe('JIRA wiki markup - Block Card', () => {
  const testCases: Array<[string, string]> = [
    [
      'should create Block Card if on its own line',
      `line 1 
      [http://...|http://...|smart-link]
      line 2 `,
    ],
    [
      'should create Block Card if nothing else in the document',
      `[http://...|http://...|smart-link]`,
    ],
    [
      'should create Block Card if nothing else in the document, file ending with new line',
      `[http://...|http://...|smart-link]
      `,
    ],
    [
      'should create Block Card if nothing else in the document, file starting with new line',
      `
      [http://...|http://...|smart-link]`,
    ],
    [
      'should create Block Card if only whitespace on the same line',
      ` [http://...|http://...|smart-link] `,
    ],
    [
      'should create Block Card if only whitespace on the same line, followed by text',
      ` [http://...|http://...|smart-link] 
      test`,
    ],
    [
      'should create Block Card if only whitespace on the same line, followed by Block Card',
      ` [http://...|http://...|smart-link] 
      [http://...|http://...|smart-link]`,
    ],
    [
      'should create Block Card if nothing else in the table cell',
      `[http://...|http://...|smart-link]
      |Heading 1|Heading 2|
      |[http://...|http://...|smart-link]|[http://...|http://...|smart-link] 
      |`,
    ],
    [
      'should create Inline Card if preceded and followed by text',
      `
      abc [http://...|http://...|smart-link] def
      `,
    ],
    [
      'should create Inline Card if preceded by text',
      `
      abc [http://...|http://...|smart-link]
      `,
    ],
    [
      'should create Inline Card if followed by text',
      `
      [http://...|http://...|smart-link] def
      `,
    ],
  ];

  for (const [testCaseDescription, markup] of testCases) {
    it(testCaseDescription, () => {
      const transformer = new WikiMarkupTransformer();
      expect(transformer.parse(markup)).toMatchSnapshot();
    });
  }
});
