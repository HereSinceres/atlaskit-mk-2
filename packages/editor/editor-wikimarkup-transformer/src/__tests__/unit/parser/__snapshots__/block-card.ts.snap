// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JIRA wiki markup - Block Card should create Block Card if nothing else in the document 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "data": null,
        "url": "http://...",
      },
      "type": "blockCard",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Block Card should create Block Card if nothing else in the document, file ending with new line 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "data": null,
        "url": "http://...",
      },
      "type": "blockCard",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Block Card should create Block Card if nothing else in the document, file starting with new line 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "data": null,
        "url": "http://...",
      },
      "type": "blockCard",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Block Card should create Block Card if nothing else in the table cell 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "data": null,
        "url": "http://...",
      },
      "type": "blockCard",
    },
    Object {
      "attrs": Object {
        "__autoSize": false,
        "isNumberColumnEnabled": false,
        "layout": "default",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "Heading 1",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "Heading 2",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "attrs": Object {
                    "data": null,
                    "url": "http://...",
                  },
                  "type": "blockCard",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "attrs": Object {
                    "data": null,
                    "url": "http://...",
                  },
                  "type": "blockCard",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Block Card should create Block Card if on its own line 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "line 1 ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "data": null,
        "url": "http://...",
      },
      "type": "blockCard",
    },
    Object {
      "content": Array [
        Object {
          "text": "line 2 ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Block Card should create Block Card if only whitespace on the same line 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "data": null,
        "url": "http://...",
      },
      "type": "blockCard",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Block Card should create Block Card if only whitespace on the same line, followed by Block Card 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "data": null,
        "url": "http://...",
      },
      "type": "blockCard",
    },
    Object {
      "attrs": Object {
        "data": null,
        "url": "http://...",
      },
      "type": "blockCard",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Block Card should create Block Card if only whitespace on the same line, followed by text 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "data": null,
        "url": "http://...",
      },
      "type": "blockCard",
    },
    Object {
      "content": Array [
        Object {
          "text": "test",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Block Card should create Inline Card if followed by text 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "data": null,
            "url": "http://...",
          },
          "type": "inlineCard",
        },
        Object {
          "text": " def",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Block Card should create Inline Card if preceded and followed by text 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "abc ",
          "type": "text",
        },
        Object {
          "attrs": Object {
            "data": null,
            "url": "http://...",
          },
          "type": "inlineCard",
        },
        Object {
          "text": " def",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Block Card should create Inline Card if preceded by text 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "abc ",
          "type": "text",
        },
        Object {
          "attrs": Object {
            "data": null,
            "url": "http://...",
          },
          "type": "inlineCard",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
